rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverWebXmlRuleProvider.java

```diff
@@ -118,9 +117,7 @@ private void addWebXmlMetadata(GraphRewrite event, EvaluationContext evaluationC
         if (StringUtils.isNotBlank(webXmlVersion)) {
             webXmlVersion = StringUtils.replace(webXmlVersion, "_", ".");
             webXml.setSpecificationVersion(webXmlVersion);
-
-            // set the tag version
-            technologyTag.setVersion(webXmlVersion);
+            technologyTagService.addTagToFileModel(xml, TECH_TAG, TECH_TAG_LEVEL, webXmlVersion);
```

Comment:
This changes the behaviour, the tech tag won't get added to the file modelif there is no webXmlVersion, whereas previously  it just didn't set the version in this case on the already-added to the file model techtag

Reply:
Absolutely, well spotted, thanks :+1: